File: Python/OOP/Inheritance/Multi-level inehritance (example 2).py
class Institute:
    __institute_name = "Institute of Engineering"  # just a random name
    __eiin = 654321

    def instituteInfo(self):
        print("Institute Name:", self.__institute_name)
        print("Institute EIIN:", self.__eiin)


class Teacher(Institute):
    teacher_name = "Guido van Rossum"
    subject = "Python Programming"

    def teacherInfo(self):
        print("Teacher Name  :", self.teacher_name)


class Student(Teacher):
    student_name = "Abdullah Al Naiem"
    id = 123456

    def studentInfo(self):
        print("Student Name  :", self.student_name)
        print("Student ID    :", self.id)
        print("Class of      :", self.subject)


class multiLevelInheritance:
    get = Student()
    get.instituteInfo()
    get.teacherInfo()
    get.studentInfo()

File: Java/OOP/Inheritance/Multi-level inehritance (example 2).java
class Institute {
    private String institute_name = "Institute of Engineering"; // just a random name
    private int eiin = 654321;

    void instituteInfo() {
        System.out.println("Institute Name: " + institute_name);
        System.out.println("Institute EIIN: " + eiin);
    }
}

class Teacher extends Institute {
    String teacher_name = "James Gosling";
    String subject = "Java Programming";

    void teacherInfo() {
        System.out.println("Teacher Name  : " + teacher_name);
    }
}

class Student extends Teacher {
    String student_name = "Abdullah Al Naiem";
    int id = 123456;

    void studentInfo() {
        System.out.println("Student Name  : " + student_name);
        System.out.println("Student ID    : " + id);
        System.out.println("Class of      : " + subject);
    }
}

class multiLevelInhertance {
    public static void main(String[] args) {
        Student get = new Student();
        get.instituteInfo();
        get.teacherInfo();
        get.studentInfo();
    }
}